src/sentry/static/sentry/app/views/dashboardsV2/detail.tsx

```diff
@@ -45,36 +35,30 @@ type Props = {
 };
 
 type State = {
-  // local state
   dashboardState: DashboardState;
   changesDashboard: DashboardListItem | undefined;
-
-  // endpoint response
-  orgDashboards: OrgDashboardResponse[] | null;
-} & AsyncComponent['state'];
-
-class DashboardDetail extends AsyncComponent<Props, State> {
+};
+class DashboardDetail extends React.Component<Props, State> {
   state: State = {
-    // AsyncComponent state
-    loading: true,
-    reloading: false,
-    error: false,
-    errors: [],
-
-    // endpoint response
-    orgDashboards: [],
-
-    // local state
     dashboardState: 'default',
     changesDashboard: undefined,
   };
 
-  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
-    const {organization} = this.props;
+  static getDerivedStateFromProps(props: Props, state: State) {
+    if (state.changesDashboard && state.changesDashboard.type === 'org') {
+      const {params} = props;
+      const dashboardId = params.dashboardId as string | undefined;
 
-    const url = `/organizations/${organization.slug}/dashboards/`;
+      if (typeof dashboardId === 'string' && state.changesDashboard.id !== dashboardId) {
+        return {
+          ...state,
+          dashboardState: 'default',
```

Comment:
Should this state be 'view', so we have view/edit modes?

Reply:
That's a better name